core/modules/replica/SqlDeleteTableJob.h

```diff
@@ -0,0 +1,148 @@
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_QSERV_REPLICA_SQLDELETETABLEJOB_H
+#define LSST_QSERV_REPLICA_SQLDELETETABLEJOB_H
+
+// System headers
+#include <functional>
+#include <list>
+#include <set>
+#include <string>
+#include <tuple>
+
+// Qserv headers
+#include "replica/SqlJob.h"
+
+// This header declarations
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+/**
+ * Class SqlDeleteTableJob represents a tool which will broadcast the same request
+ * for deleting an existing table from all worker databases of a setup. Result sets
+ * are collected in the above defined data structure.
+ */
+class SqlDeleteTableJob : public SqlJob {
+public:
+    /// The pointer type for instances of the class
+    typedef std::shared_ptr<SqlDeleteTableJob> Ptr;
+
+    /// The function type for notifications on the completion of the request
+    typedef std::function<void(Ptr)> CallbackType;
+
+    /// @return the unique name distinguishing this class from other types of jobs
+    static std::string typeName();
+
+    /**
+     * Static factory method is needed to prevent issue with the lifespan
+     * and memory management of instances created otherwise (as values or via
+     * low-level pointers).
+     *
+     * @param database
+     *   the name of a database from which a table will be deleted
+     *
+     * @param table
+     *   the name of an existing table to be deleted
+     *
+     * @param allWorkers
+     *   engage all known workers regardless of their status. If the flag
+     *   is set to 'false' then only 'ENABLED' workers which are not in
+     *   the 'READ-ONLY' state will be involved into the operation.
+     *
+     * @param controller
+     *   is needed launching requests and accessing the Configuration
+     *
+     * @param parentJobId
+     *   (optional) identifier of a parent job
+     *
+     * @param onFinish
+     *   (optional) callback function to be called upon a completion of the job
+     *
+     * @param options
+     *   (optional) defines the job priority, etc.
+     *
+     * @return
+     *   pointer to the created object
+     */
+    static Ptr create(std::string const& database,
+                      std::string const& table,
+                      bool allWorkers,
+                      Controller::Ptr const& controller,
+                      std::string const& parentJobId=std::string(),
+                      CallbackType const& onFinish=nullptr,
+                      Job::Options const& options=defaultOptions());
+
+    // Default construction and copy semantics are prohibited
+
+    SqlDeleteTableJob() = delete;
+    SqlDeleteTableJob(SqlDeleteTableJob const&) = delete;
+    SqlDeleteTableJob& operator=(SqlDeleteTableJob const&) = delete;
+
+    ~SqlDeleteTableJob() final = default;
```

Comment:
There are several destructors marked as final and they effectively make the class final. I think it's better to label the class as final.

Reply:
Good suggestion!